options.go

```diff
@@ -53,6 +58,17 @@ func (o *opts) apply(opts *opts) {
 	opts.cleanup = o.cleanup
 }
 
+// validate the options.
+func (o *opts) validate() error {
+	if o.maxRetries < 0 {
+		return errors.New("maxRetryAttempts should be greater than 0")
+	}
+	if o.maxSleep < time.Duration(0) {
```

Comment:
If the error message below is accurate, this check should probably be a `<=`.
```suggestion
	if o.maxSleep <= 0 {
```

Otherwise the error message should probably say "should be 0s or greater"?